### PR TITLE
Refactored handlers

### PR DESCRIPTION
**Pull request by dolgopolovwork**
Bitcoin withdrawal service refactor. Handlers implement the same interface `SetAccountDetailHandler`. 
Bitcoin address generation service refactor is coming soon.

### CODE REVIEW

File: btc-withdrawal/src/main/kotlin/com/d3/btc/withdrawal/init/BtcWithdrawalInitialization.kt [hunk 1]
```diff
@@ -120,39 +113,15 @@ class BtcWithdrawalInitialization(
...
-            }
-        // Handle 'set new consensus' events
-        getSetDetailCommands(block).filter { command -> isNewConsensus(command) }
-            .forEach { command ->
-                newConsensusDataHandler.handleNewConsensusCommand(command.setAccountDetail)
+        // Handle other commands
+        getSetDetailCommands(block).map { it.setAccountDetail }
+            .forEach { setAccountDetailCommand ->
+                accountDetailHandlers.forEach { handler ->
+                    if (handler.filter(setAccountDetailCommand)) {
```
Comment (Mingela) [inline btc-withdrawal/src/main/kotlin/com/d3/btc/withdrawal/init/BtcWithdrawalInitialization.kt]: maybe it is better to make `filter` a part of `handle`? just not to forget to `filter` somewhere else

File: btc-withdrawal/src/main/kotlin/com/d3/btc/withdrawal/init/BtcWithdrawalInitialization.kt [hunk 2]
```diff
@@ -51,13 +49,9 @@ class BtcWithdrawalInitialization(
...
     private val btcNetworkConfigProvider: BtcNetworkConfigProvider,
-    private val newSignatureEventHandler: NewSignatureEventHandler,
-    private val newBtcClientRegistrationHandler: NewBtcClientRegistrationHandler,
     private val newTransferHandler: NewTransferHandler,
-    private val newChangeAddressHandler: NewChangeAddressHandler,
-    private val newConsensusDataHandler: NewConsensusDataHandler,
-    private val newTransactionCreatedHandler: NewTransactionCreatedHandler,
-    private val broadcastTransactionHandler: BroadcastTransactionHandler,
+    @Qualifier("withdrawalHandlers")
+    private val accountDetailHandlers: List<SetAccountDetailHandler>,
```
Comment (Mingela) [inline btc-withdrawal/src/main/kotlin/com/d3/btc/withdrawal/init/BtcWithdrawalInitialization.kt:54]: how about more abstraction `List -> Iterable`?

File: btc-withdrawal/src/test/java/com/d3/btc/withdrawal/handler/NewConsensusDataHandlerTest.java [hunk 3]
```diff
@@ -80,7 +80,7 @@ public void testHandleNewConsensusCommandHasNotBeenEstablished() {
                 .setAccountId("test@" + BTC_CONSENSUS_DOMAIN)
                 .setValue(withdrawalConsensus.toJson())
                 .build();
-        newConsensusDataHandler.handleNewConsensusCommand(newConsensusCommand);
+        newConsensusDataHandler.handle(newConsensusCommand);
         verify(withdrawalTransferService).withdraw(any(), any());
     }
 }
```
Comment (Mingela) [inline btc-withdrawal/src/test/java/com/d3/btc/withdrawal/handler/NewConsensusDataHandlerTest.java]: 😨

File: btc/src/main/kotlin/com/d3/btc/handler/SetAccountDetailHandler.kt [hunk 4]
```diff
@@ -0,0 +1,21 @@
...
+ */
+interface SetAccountDetailHandler {
+    /**
+     * Handling logic
+     * @param command - command to handle
+     */
+    fun handle(command: Commands.SetAccountDetail)
+
+    /**
+     * Filter of commands. Only filtered commands will be handled
```
Comment (Mingela) [inline btc/src/main/kotlin/com/d3/btc/handler/SetAccountDetailHandler.kt:31]: `Only filtered commands will be handled` that is not true for now 😆 
You can refactor the method to `isProcessable` (if you need it accessable at all) and use it inside `handle`

**Review by Mingela — APPROVED** [review]
please see the comments as well

**Review by Alexey-N-Chernyshov — APPROVED** [review]

**Pull request merged**